Marlin/src/feature/leds/neopixel.cpp

```diff
@@ -36,10 +36,20 @@
 
 Adafruit_NeoPixel pixels(NEOPIXEL_PIXELS, NEOPIXEL_PIN, NEOPIXEL_TYPE + NEO_KHZ800);
 
+#ifdef NEOPIXEL_BKGD_LED_INDEX
+  void set_neopixel_color_background() {
+    uint8_t background_color[4] = NEOPIXEL_BKGD_COLOR;
+    pixels.setPixelColor(NEOPIXEL_BKGD_LED_INDEX, pixels.Color(background_color[0], background_color[1], background_color[2], background_color[3]));
+  }
+#endif
+
 void set_neopixel_color(const uint32_t color) {
   for (uint16_t i = 0; i < pixels.numPixels(); ++i) {
     #ifdef NEOPIXEL_BKGD_LED_INDEX
-      if (NEOPIXEL_BKGD_LED_INDEX == i) i++;
+      if (i == NEOPIXEL_BKGD_LED_INDEX && color != 0x000000) {
+        set_neopixel_color_background();
+        continue;
+      }
```

Comment:
This seems like it would break the previous functionality, where the loop skips the background LED index. As this is here, *any* call to `set_neopixel_color` will now update the background of the LCD.

Q: Is this logic needed for all cases of `NEOPIXEL_BKGD_LED_INDEX` or just for `LED_BACKLIGHT_TIMEOUT`?

Reply:
Only for LED_BACKLIGHT_TIMEOUT, It is necessary to turn off the background LED. Setting any color does not affect the functionality only when the color is off / black. This miodification affects if it is ordered to turn off the LEDs (panel or gcode), the display LED will also turn off, not for the color change. Maybe something like this would be better:

  #ifdef LED_BACKLIGHT_TIMEOUT
	if (i == NEOPIXEL_BKGD_LED_INDEX && color != 0x000000) {
	  set_neopixel_color_background();
	  continue;
	}
  #else
	if (i == NEOPIXEL_BKGD_LED_INDEX {
	  set_neopixel_color_background();
	  continue;
	}
  #endif